tools/js-sdk-release-tools/packages/typescript-codegen-breaking-change-detector/src/azure/diff/declaration-diff.ts

```diff
@@ -0,0 +1,433 @@
+    if (
+      (isPropertyMethod(targetProperty) || isPropertyArrowFunction(targetProperty)) &&
+      (isPropertyMethod(sourceProperty) || isPropertyArrowFunction(sourceProperty))
+    ) {
+      const functionPropertyDetails = findFunctionPropertyBreakingChangeDetails(sourceProperty, targetProperty);
+      return [...result, ...functionPropertyDetails];
+    }
+
+    throw new Error('Should never reach here');
+  }, new Array<DiffPair>());
+  return [...removed, ...changed];
+}
+
+function findReturnTypeBreakingChangesCore(source: Node, target: Node): DiffPair[] {
+  const reasons = findBreakingReasons(source, target);
+  if (reasons === DiffReasons.None) return [];
+  const getName = (node: Node) => (Node.hasName(node) ? node.getName() : node.getText());
+  const targetNameNode = target ? { name: getName(target), node: target } : undefined;
+  const sourceNameNode = source ? { name: getName(source), node: source } : undefined;
+  const pair = createDiffPair(DiffLocation.Signature_ReturnType, reasons, sourceNameNode, targetNameNode);
+  return [pair];
+}
+
+function findReturnTypeBreakingChanges(sourceMethod: Symbol, targetMethod: Symbol): DiffPair[] {
+  const targetDeclaration = targetMethod.getValueDeclarationOrThrow();
+  const sourceDeclaration = sourceMethod.getValueDeclarationOrThrow();
+  return findReturnTypeBreakingChangesCore(sourceDeclaration, targetDeclaration);
+}
+
+function findParameterBreakingChangesCore(
+  sourceParameters: ParameterDeclaration[],
+  targetParameters: ParameterDeclaration[],
+  sourceName: string,
+  targetName: string,
+  sourceNode: Node,
+  targetNode: Node
+): DiffPair[] {
+  const pairs: DiffPair[] = [];
+
+  // handle parameter counts
+  const isSameParameterCount = targetParameters.length === sourceParameters.length;
+  if (!isSameParameterCount) {
+    const source = { name: sourceName, node: sourceNode };
+    const target = { name: targetName, node: targetNode };
+    const pair = createDiffPair(DiffLocation.Signature_ParameterList, DiffReasons.CountChanged, source, target);
+    pairs.push(pair);
+    return pairs;
+  }
+
+  // NOTE: parameter count is the same
+  // handle each parameter
+  targetParameters.forEach((targetParameter, i) => {
+    const sourceParameter = sourceParameters[i];
+    const getParameterNameNode = (p: ParameterDeclaration | undefined) =>
+      p ? { name: p.getName() || '', node: p } : undefined;
+    const target = getParameterNameNode(targetParameter);
+    const source = getParameterNameNode(sourceParameter);
+    const reasons = findBreakingReasons(sourceParameter, targetParameter);
+    const pair = createDiffPair(DiffLocation.Parameter, reasons, source, target);
+    pair.reasons = findBreakingReasons(sourceParameter, targetParameter);
+    if (pair.reasons !== DiffReasons.None) pairs.push(pair);
+  });
+
+  return pairs;
+}
+
+// TODO: not support for overloads
+function findParameterBreakingChanges(sourceMethod: Symbol, targetMethod: Symbol): DiffPair[] {
```

Comment:
I wonder why I see the changelog for isUnexpected helper overloads change if it's not supported yet?

Reply:
this function is used for overloads inside a interface. 
isUnexpected is a function out of interface.